contracts/swaps/SwapsUser.sol

```diff
@@ -124,24 +127,45 @@ contract SwapsUser is State, SwapsEvents, FeesHelper {
         } else {
             require(vals[0] <= vals[1], "min greater than max");
         }
-
-        require(loanDataBytes.length == 0, "invalid state");
-        (destTokenAmountReceived, sourceTokenAmountUsed) = _swapsCall_internal(
-            addrs,
-            vals
-        );
-
+        if (loanDataBytes.length == 0) {
+            (
+                destTokenAmountReceived,
+                sourceTokenAmountUsed
+            ) = _swapsCall_internal(addrs, vals, "");
+        } else {
+            (uint128 flags, bytes[] memory payload) = abi.decode(
+                loanDataBytes,
+                (uint128, bytes[])
+            );
+            if (flags & DEX_SELECTOR_FLAG != 0) {
+                (
+                    destTokenAmountReceived,
+                    sourceTokenAmountUsed
+                ) = _swapsCall_internal(addrs, vals, payload[0]);
+            } else {
+                (
+                    destTokenAmountReceived,
+                    sourceTokenAmountUsed
+                ) = _swapsCall_internal(addrs, vals, "");
+            }
+        }
         if (vals[2] == 0) {
             // there's no minimum destTokenAmount, but all of vals[0] (minSourceTokenAmount) must be spent, and amount spent can't exceed vals[0]
-            require(sourceTokenAmountUsed == vals[0], "swap too large to fill");
+            require(
+                sourceTokenAmountUsed == vals[0],
+                "swap too large to fill test"
```

Comment:
whats `test`?

Reply:
Was part of testing dex selector initially, word "test" has been removed.